concrete/src/Package/Package.php

```diff
@@ -733,219 +737,70 @@ public function upgrade()
-     */
-    public function getMetadataDriver()
-    {
-        if ($this->metadataDriver === self::PACKAGE_METADATADRIVER_ANNOTATION){
-            if(version_compare($this->getApplicationVersionRequired(), '5.8.0', '<')){
-                // Legacy - uses SimpleAnnotationReader
-                $cachedSimpleAnnotationReader = $this->app->make('orm/cachedSimpleAnnotationReader');
-                $simpleAnnotationDriver = new \Doctrine\ORM\Mapping\Driver\AnnotationDriver($cachedSimpleAnnotationReader, $this->getPackageMetadataPaths());
-                return $simpleAnnotationDriver;
-            }else{
-                // Use default AnnotationReader
-                $cachedAnnotationReader = $this->app->make('orm/cachedAnnotationReader');
-                $annotationDriver = new \Doctrine\ORM\Mapping\Driver\AnnotationDriver($cachedAnnotationReader, $this->getPackageMetadataPaths());
-                return $annotationDriver;
-            }
-        } else if ($this->metadataDriver === self::PACKAGE_METADATADRIVER_XML){
-            $driverImpl = new \Doctrine\ORM\Mapping\Driver\XmlDriver($this->getPackageMetadataPaths());
-
-        } else if ($this->metadataDriver === self::PACKAGE_METADATADRIVER_YAML){
-            $driverImpl = new \Doctrine\ORM\Mapping\Driver\YamlDriver($this->getPackageMetadataPaths());
-        }
-        return $driverImpl;
-    }
-    
-    /**
-     * Get path to the location containing the metadata info
-     * 
-     * @return array 
-     */
-    public function getPackageMetadataPaths()
-    {
-        // annotations entity path
-        $paths = array();
-        if ($this->metadataDriver === self::PACKAGE_METADATADRIVER_ANNOTATION){
-            // Support for the legacy method for backwards compatibility
-            $paths = $this->getPackageEntityPaths();
-        } else if ($this->metadataDriver === self::PACKAGE_METADATADRIVER_XML){
-            // return xml metadata dir
-            $paths =  array($this->getPackagePath() . DIRECTORY_SEPARATOR . REL_DIR_METADATA_XML);
-        } else if ($this->metadataDriver === self::PACKAGE_METADATADRIVER_YAML){
-            // return yaml metadata dir
-            $paths =  array($this->getPackagePath() . DIRECTORY_SEPARATOR . REL_DIR_METADATA_YAML);
-        }
-
-        // Add additional source path to the default namespace
-        // if pkgAutoloaderMapCoreExtensions is true
-        $corePath = !$this->enableLegacyNamespace() ? $this->getPackagePath() . DIRECTORY_SEPARATOR . DIRNAME_CLASSES . DIRECTORY_SEPARATOR . 'Concrete' : '';
-        if(!empty($corePath)){
-            $paths[] = $corePath;
-        }
-
-        $returnPaths = array();
-        foreach($paths as $path) {
-            if (is_dir($path)) {
-                $returnPaths[] = $path;
-            }
+        $em = $this->getPackageEntityManager();
+        if (is_object($em)) {
+            $this->destroyProxyClasses($em);
+            $this->installEntitiesDatabase();
```

Comment:
I know we want to keep backwards compatibility but this long stripe of red makes me so happy